python/pyspark/sql/pandas/serializers.py

```diff
@@ -90,7 +90,10 @@ def load_stream(self, stream):
         import pyarrow as pa
         reader = pa.ipc.open_stream(stream)
         for batch in reader:
-            yield batch
+            split_batch = pa.RecordBatch.from_arrays([
+                pa.concat_arrays([array]) for array in batch
+            ], schema=batch.schema)
```

Comment:
Sorry for asking a question without taking a close look but would you mind elaborating why we should do this?

Reply:
I added a comment, but please let me know if it's unclear.

Reply:
This looks a little strange to me too. Is `concat_arrays` actually doing anything here, and if so, wouldn't it do it for the case that `selfDestruct` is False too?